### PR TITLE
FISH-1497 Improve deploying an application from a Maven repository feature to support snapshot version

### PR DESCRIPTION
**Pull request by kalinchan**
## Description

If version ends with SNAPSHOT, get maven-metadata.xml from maven repo. Get value which is latest SNAPSHOT version 

### Testing Performed

Manual testing with local repo and payara nexus repo. 

Asadmin command for nexus repo 
<h5>asadmin deploy-remote-archive --additionalRepositories https://nexus.payara.fish/repository/payara-snapshots fish.payara.extras,ejb-http-client,5.2021.5-SNAPSHOT<h5>

### Testing Environment

openjdk version "1.8.0_292"
ubunutu 20.04
maven 3.6.3

### CODE REVIEW

**Comment by kalinchan** [comment]
jenkins test please

**Review by Pandrex247 — CHANGES_REQUESTED** [review]
Haven't reviewed functionality yet, but from eyeballing...

**Review by lprimak — CHANGES_REQUESTED** [review]
Please change all occurrences of `String + String + String` to `String.format()` for improved readability

File: nucleus/deployment/common/src/main/java/fish/payara/deployment/util/GAVConvertor.java [hunk 1]
```diff
@@ -135,11 +146,53 @@ private GAVConvertor() {
...
+                    Node node = nodeList.item(0);
+                    Element element = (Element) node;
+                    //Get latest snapshot version
+                    String version = element.getElementsByTagName("value").item(0).getTextContent();
+                    artefactFileName = GAVMap.get("artefactId") + "-" + version;
+                    logger.log(Level.FINE, "Found version {0} from maven-metadata.xml", version);
+                    break;
+                }
+            } catch (URISyntaxException e) {
+                logger.log(Level.WARNING, "Error creating maven metadata URI");
```
Comment (lprimak) [inline nucleus/deployment/common/src/main/java/fish/payara/deployment/util/GAVConvertor.java:182]: this should be combined into a single catch statement i.e. 
```
catch (URISyntaxExceptino | IOExceptino | Parser... e) {}
```

Comment (MarkWareham) [reply nucleus/deployment/common/src/main/java/fish/payara/deployment/util/GAVConvertor.java:182]: That would mean that the log statements would need to be the same.

**Review by MarkWareham — APPROVED** [review]
> Please change all occurrences of String + String + String to String.format() for improved readability

A minor issue that should not prevent merge. I would say it should be "approved" with advised, but not necessary changes

**Review by MeroRai — APPROVED** [review]

**Comment by eliotmartin** [comment]
jenkins test please

**Pull request merged**